pkg/datasources/dynamic_tables.go

```diff
@@ -0,0 +1,293 @@
+package datasources
+
+import (
+	"context"
+	"database/sql"
+	"log"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var dynamicTablesSchema = map[string]*schema.Schema{
+	"like": {
```

Comment:
Is this the first datasource using like + in + startsWith + limit? (I've searched a bit, and I see that database datasource - `pkg/datasource/databases.go` - has both `pattern` and `starts_with`)

I am asking because definitions for these four parameters should be the same (or almost the same) throughout our codebase. Maybe we could extract their descriptions? Or even extract `Schema`s for them (I mean for like, in, etc)?

This shouldn't be done inside this PR of course, but I am putting it here as a food for thought.